src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,246 @@
+		this.accessToken = null; /* ACCESSTOKEN */
+		EduSharingConnector.instance = this;
+	}
+
+	static get headers() {
+		return {
+			Accept: 'application/json',
+			'Content-type': 'application/json',
+		};
+	}
+
+	static get authorization() {
+		const headers = {
+			...EduSharingConnector.headers,
+			Authorization: `Basic ${Buffer.from(`${Configuration.get('ES_USER')
+			}:${Configuration.get('ES_PASSWORD')}`).toString(
+				'base64',
+			)}`,
+		};
+
+		return headers;
+	}
+
+
+	// gets cookie (JSESSION) and attach it to header
+	getCookie() {
+		const cookieOptions = {
+			uri: `${Configuration.get('ES_DOMAIN')}${ES_PATH.AUTH}`,
+			method: 'GET',
+			headers: EduSharingConnector.authorization,
+			resolveWithFullResponse: true,
+			json: true,
+		};
+		return request(cookieOptions)
+			.then((result) => {
+				if (
+					result.statusCode !== 200
+					|| result.body.isValidLogin !== true
+				) {
+					throw Error('authentication error with edu sharing');
+				}
+				return result.headers['set-cookie'][0];
+			})
+			.catch((err) => {
+				logger.error('error: ', err);
+			});
+	}
+
+	// gets access_token and refresh_token
+	getAuth() {
+		const oauthoptions = {
+			method: 'POST',
+			url: `${Configuration.get('ES_DOMAIN')}${ES_PATH.TOKEN}`,
+			headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+
+			body: `grant_type=${Configuration.get('ES_GRANT_TYPE')}&client_id=${
+				Configuration.get('ES_CLIENT_ID')
+			}&client_secret=${Configuration.get('ES_OAUTH_SECRET')}&username=${
+				Configuration.get('ES_USER')
+			}&password=${Configuration.get('ES_PASSWORD')}`,
+			timeout: REQUEST_TIMEOUT,
+		};
+		return request(oauthoptions).then((result) => {
+			if (result) {
+				const parsedResult = JSON.parse(result);
+				return Promise.resolve(parsedResult.access_token);
+			}
+			return Promise.reject(new GeneralError('Oauth failed'));
+		});
+	}
+
+	allConfigurationValuesHaveBeenDefined() {
+		return (
+			Configuration.has('ES_DOMAIN')
+			&& Configuration.has('ES_USER')
+			&& Configuration.has('ES_PASSWORD')
+			&& Configuration.has('ES_GRANT_TYPE')
+			&& Configuration.has('ES_OAUTH_SECRET')
+			&& Configuration.has('ES_CLIENT_ID')
+		);
+	}
+
+	async requestRepeater(options) {
+		let retry = 0;
+		const errors = [];
+		do {
+			try {
+				const eduResponse = await request(options);
+				return JSON.parse(eduResponse);
+			} catch (e) {
+				if (e.statusCode === 401) {
+					logger.info(`Trying to renew Edu Sharing connection. Attempt ${retry}`);
+					await this.login();
+				} else {
+					logger.error(e);
+					errors.push(e);
+				}
+			}
+			retry += 1;
+		} while (retry < 3);
+
+		throw new GeneralError('Edu Sharing Retry failed', errors);
+	}
+
+
+	async login() {
+		this.authorization = await this.getCookie();
+		this.accessToken = await this.getAuth();
+	}
+
+	isLoggedin() {
+		// returns false if cookie or accesstoken is falsy
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async GET(id, params) {
```

Comment:
Codacy found an issue: [Block must not be padded by blank lines.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4977295)